File: Python/EjemploWhile/EjemploWhile.py
def ingresar_numeros():
    numeros = []
    continuar = True

    while continuar:
        try:
            entrada = input("Ingrese un número (0 para terminar): ")
            numero = int(entrada)

            if numero == 0:
                continuar = False
            else:
                numeros.append(numero)
        except ValueError:
            print("Entrada inválida. Por favor, ingrese un número válido.")

    return numeros

if __name__ == "__main__":
    print("Ingresar Números (Ingrese 0 para terminar)")
    print("========================================")
    print()

    numeros = ingresar_numeros()

    print()
    print(f"Cantidad de números ingresados: {len(numeros)}")

    print()
    input("Presione cualquier tecla para salir.")

File: Java/EjemploWhile/src/EjemploWhile.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class EjemploWhile {
    public static void main(String[] args) {
        System.out.println("Ingresar Números (Ingrese 0 para terminar)");
        System.out.println("========================================");
        System.out.println();

        List<Integer> numeros = ingresarNumeros();

        System.out.println();
        System.out.println("Cantidad de números ingresados: " + numeros.size());

        System.out.println();
        System.out.println("Presione cualquier tecla para salir.");
        new Scanner(System.in).nextLine();
    }

    static List<Integer> ingresarNumeros() {
        List<Integer> numeros = new ArrayList<>();
        int numero;
        boolean continuar = true;

        while (continuar) {
            System.out.print("Ingrese un número (0 para terminar): ");
            Scanner scanner = new Scanner(System.in);
            String input = scanner.nextLine();

            try {
                numero = Integer.parseInt(input);

                if (numero == 0) {
                    continuar = false;
                } else {
                    numeros.add(numero);
                }
            } catch (NumberFormatException e) {
                System.out.println("Entrada inválida. Por favor, ingrese un número válido.");
            }
        }

        return numeros;
    }
}
